emily/cdk/lib/emily-stack.ts

```diff
@@ -0,0 +1,196 @@
+            tableName: EmilyStackUtils.getResourceName(tableId, props),
+            partitionKey: {
+                name: 'BlockHeight',
+                type: dynamodb.AttributeType.NUMBER,
+            },
+            sortKey: {
+                name: 'BlockHash',
+                type: dynamodb.AttributeType.BINARY,
+            }
+        });
+    }
+
+    createOrUpdateOperationLambda(
+        depositTable: dynamodb.Table,
+        withdrawalTable: dynamodb.Table,
+        chainstateTable: dynamodb.Table,
+        props: EmilyStackProps
+    ): lambda.Function {
+
+        // This must match the Lambda name from the @aws.apigateway#integration trait in the
+        // smithy operations and resources that should be handled by this Lambda.
+        const operationLambdaId: string = "OperationLambda";
+
+        const operationLambda: lambda.Function = new lambda.Function(this, operationLambdaId, {
+            functionName: EmilyStackUtils.getResourceName(operationLambdaId, props),
+            architecture: lambda.Architecture.ARM_64, // <- Will need to change when run locally for x86
+            runtime: lambda.Runtime.PROVIDED_AL2023,
+            code: lambda.Code.fromAsset(EmilyStackUtils.getPathFromProjectRoot(
+                "target/lambda/emily-operation-lambda/bootstrap.zip"
+            )),
+            // Lambda should be very fast. Something is wrong if it takes > 5 seconds.
+            timeout: cdk.Duration.seconds(30),
+            handler: "main",
+            environment: {
+                // Give lambda access to the table name.
+                DEPOSIT_TABLE_NAME: depositTable.tableName,
+                WITHDRAWAL_TABLE_NAME: withdrawalTable.tableName,
+                CHAINSTATE_TABLE_NAME: chainstateTable.tableName,
+                // Declare an environment variable that will be overwritten in local SAM
+                // deployments the AWS stack. SAM can only set environment variables that are
+                // already expected to be present in the lambda.
+                IS_LOCAL: "false",
+            }
+        });
+
+        // Give the server lambda full access to the DynamoDB table.
+        depositTable.grantReadWriteData(operationLambda);
+        withdrawalTable.grantReadWriteData(operationLambda);
+        chainstateTable.grantReadWriteData(operationLambda);
+
+        // Return lambda resource.
+        return operationLambda;
+    }
+
+    createOrUpdateApi(
+        serverLambda: lambda.Function,
+        props: EmilyStackProps
+    ): apig.SpecRestApi {
+
+        const restApiId: string  = "EmilyAPI";
+        const restApi: apig.SpecRestApi = new apig.SpecRestApi(this, restApiId, {
+            restApiName: EmilyStackUtils.getResourceName(restApiId, props),
+            apiDefinition: EmilyStackUtils.restApiDefinition(EmilyStackUtils.getPathFromProjectRoot(
+                ".generated-sources/openapi/openapi/Emily.openapi.json"
+            )),
```

Comment:
Very cool.

Hmm, is there a way for us to ensure that this is built (or specifically rebuilt if it's outdated) before invoking the CDK?

Reply:
Good callout! I'll have the npm build command trigger the building of the generated sources.

Reply:
Done.